babi/prompt.py

```diff
@@ -14,12 +16,20 @@
 
 
 class Prompt:
-    def __init__(self, screen: Screen, prompt: str, lst: list[str]) -> None:
+    def __init__(
+        self,
+        screen: Screen,
+        prompt: str, lst: list[str],
+        file_glob: bool = False,
+    ) -> None:
         self._screen = screen
         self._prompt = prompt
         self._lst = lst
         self._y = len(lst) - 1
         self._x = len(self._s)
+        self._dispatch = Prompt.DISPATCH.copy()
+        if file_glob:
+            self._dispatch[b'^I'] = Prompt._file_complete
```

Comment:
unknown keys are already ignored in prompts -- the completion callback should just check if it's enabled or not rather than having to copy this dict every time

Reply:
I don't think I  understand this point - a key seems to be considered "known" as soon as it's in the dispatch dict, so I can't just add it globally.

If I didn't copy the dict, I'd be modifying the global `DISPATCH` dict, AFAIUI?

Also, for the callback to check on whether globbing is enabled, i'd have to make `file_glob` an instance variable, correct?

Reply:
I'm saying you should handle the option inside the callback